docs/Developers/Debug_Bears.rst

```diff
@@ -0,0 +1,77 @@
+Debug Bears
+===========
+
+This document provide a brief overview of coala's Debugging Interface.
+The Debugging Interface will help users to debug Bear code and step through
+it using pdb interface
+
+Usage of coala's Debugger
+=========================
+
+Command Line Interface
+----------------------
+
+Users can specify the bear they want to debug using ``--debug-bears``, i.e.
+
+.. code:: shell
+
+    $coala -b PEP8Bear,HelloWorldBear -f <filename> --debug-bears HelloWorldBear
+
+Below is the code given for a simple ``HelloWorldBear`` that prints a debug
+message for each file:
+
+.. code:: python
+
+    import logging
+
+    from coalib.bears.LocalBear import LocalBear
+
+
+    class HelloWorldBear(LocalBear):
+        def run(self,
+                filename,
+                file):
+            logging.info('Please follow newcomer guide.')
+            logging.debug('Hello World! Checking file {}.'.format(filename))
+
+After invoking the coala's debugger it will step into the ``run()`` method of a
+HelloWorldBear and help users to debug Bear code with passed settings.
+
+If bear names are not specified for ``--debug-bears`` then it will by
+default debug all the bears passed through ``--bears`` or ``-b`` argument.
+
+.. code:: shell
+
+    $ coala --bears HelloWorldBear -files <filename> --debug-bears
+
+.. note::
+
+    A bear may depend on results from different bears. The debugger will debug
+    all bears on which a bear is dependent on as well.
```

Comment:
I'm not sure if we should clarify, that it depends on how you use `--debug-bears`. Without arguments, everything is debugged, but if you write specific bears (e.g. `--debug-bears MyBear`), then dependencies won't be debugged.

Reply:
Okay I am sure, we should clarify that :sweat_smile:

Reply:
I have added that also in my code in the case where `--debug-bears MyBear` then Debugger will also debug the dependency Bear of `MyBear`. :P so for both the case it will debug :P